release/long_running_tests/workloads/serve_failure.py

```diff
@@ -30,6 +30,11 @@
 
 RAY_UNIT_TEST = "RAY_UNIT_TEST" in os.environ
 
+# RAY_OVERRIDE_RESOURCES will prevent "num_cpus" working
+# when adding customized resource node to Cluster.
+if "RAY_OVERRIDE_RESOURCES" in os.environ:
+    del os.environ["RAY_OVERRIDE_RESOURCES"]
```

Comment:
Could you add an assertion inside the `RandomKiller` to make sure the env var isn't set? That would let us catch any regressions in the testing setup.